File: server/Models/Vault.cs
using System.ComponentModel.DataAnnotations;

namespace keepr.Models;

public class Vault : DbItem<int>
{
  [MinLength(1), MaxLength(50)] public string Name { get; set; }
  [MinLength(1), MaxLength(1000)] public string Description { get; set; }
  [MaxLength(500), Url] public string Img { get; set; }
  public bool? IsPrivate { get; set; }
  public string CreatorId { get; set; }
  public Profile Creator { get; set; }
}

File: client/src/models/Vault.js
import { DbItem } from "./DbItem.js";
import { Profile } from "./Profile.js";

export class Vault extends DbItem{
  constructor(data){
    super(data);
    this.name = data.name;
    this.description = data.description;
    this.img = data.img;
    this.isPrivate = data.isPrivate;
    this.creatorId = data.creatorId;
    this.creator = new Profile(data.creator);
  }
}